gms/gossiper.cc

```diff
@@ -1174,6 +1173,12 @@ future<> gossiper::evict_from_membership(inet_address endpoint, permit_id pid) {
     auto permit = co_await lock_endpoint(endpoint, pid);
     co_await mutate_live_and_unreachable_endpoints([endpoint] (gossiper& g) {
         g._unreachable_endpoints.erase(endpoint);
+        // When evict_from_membership is called, the endpoint should have
+        // already been erased from _live_endpoints, so enforce that here.
```

Comment:
Are we 100% sure that must indeed be the case?

I'm imagining a decommissioning node, let's say it successfully decommissioned and shutdown, and now we evict it from membership before we managed to mark it as dead. I don't think that would be a problem, the two events can happen "at the same time".

Reply:
`evict_from_membership` is called either explicitly after `remove_endpoint`, which erases the endpoint from live_endpoints, or after the endpoint has expired in _expire_time_endpoint_map.
It inserted there by `advertise_token_removed` -> `add_expire_time_for_endpoint` which happens when the node has left the ring, so it's not supposed to be alive in this state.

That said, we don't have to strict in this case and just erase it unconditionally from _live_endpoints and not care if it was there or not.